### ISSUE TITLE
Ingredient based recipe selection feature

### ISSUE THREAD

**Issue by rubikscuber** [flauschtrud/broccoli#195]
As a user I would like to be able to select a few ingredients that I have at home and get a filter for all the recipes that include this list, so that I can find recipes to cook without the need to search each ingredient after the other.

There is a nice implementation done in the drinkable app - see second and third screenshot: [drinkable app fdroid](https://f-droid.org/en/packages/com.moimob.drinkable/)

**Comment by flauschtrud** [comment]
You can actually search for multiple ingredients at once. The native search that comes with the database is actually quite good (although it unfortunately can't find prefixes and suffixes at its current state).

The title, description, URL and ingredient list are indexed. So you can search for multiple ingredients, specific URLs in combination with ingredients, hashtags and so on. You can also filter by category first and then do your search.

So you can already accomplish quite a lot without the need for an ingredient management system. Does this help with your usecase?

**Comment by flauschtrud** [comment]
Yes, the synonyms are tough. Theoretically the search index could be built including synonyms but I think that does not make sense without accessing an online service which provides the synonyms.

Regarding the hashtags: I just created an issue for collecting ideas for improving stuff like this #197 Would be happy to hear if you have anything more to add to this.
